rollup/pipeline/pipeline.go

```diff
@@ -310,15 +310,20 @@ func (p *Pipeline) cccStage(candidates <-chan *BlockCandidate, deadline time.Tim
 	var deadlineReached bool
 
 	go func() {
+		deadlineTimer := time.NewTimer(time.Until(deadline))
 		defer func() {
 			close(resultCh)
+			if !deadlineReached && !deadlineTimer.Stop() {
```

Comment:
I think we can just call `deadlineTimer.Stop` without the check and draining. The docs are a bit misleading in saying that you need to drain the channel when calling `Stop`. It's only needed when resetting a channel. There's a discussion about it here: https://github.com/golang/go/issues/27169#issuecomment-475221204.

Reply:
Nice, looks better.